### DISCUSSION TITLE
Tutorial 2: Stateful Firewall and Bruteforce SSH Attempts

### DISCUSSION THREAD

**Discussion by nyameko** [chpc-tech-eval/scc#102] [Show and tell]
Provided that you **DO NOT** enable password authentication to your head node's SSH server, there is little threat from Brute-force SSH attacks, but dropping all these queries nonetheless utilizes processing power from your head node.

You are advised to implement a stateful firewall and explore further methods to mitigate threats against your public facing interfaces.

Here is a sample of some of the incessant attempts:

 
![ssh_bruteforce_attacks](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/21206557/73934b48-205a-4daf-bc43-d7cebaa8b5a1)

**Comment by Talhahpatelia** [comment]
### Wits-A

Here is an image of the incessant attempts:
<img width="960" alt="bruteforce" src="https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/29210701/0ba3b4df-6af3-49f7-90a7-5b09d9cfe2c3">

**Reply by n-zwane** [reply]
Technocrats:
<img width="1920" height="1080" alt="firewall" src="https://github.com/user-attachments/assets/1bdc2f4a-f1de-4b15-909a-424c2d9e1f55" />

**Comment by CraftierLamb599** [comment]
# Wits-1
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/137203651/2f613555-45f5-4f4e-865b-b8d4952d8f7e)

**Comment by Kohzmik** [comment]
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/138365090/5ce1a467-c48e-4854-adfa-3dd090ecb42f)
CUT_Team

**Comment by prince-taker** [comment]
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/69721467/823b6ce3-571b-4671-a118-cb6595181ed3)
Ctrl Alt Elite

**Comment by Cyphers-UFH** [comment]
Cyphers - UFH

![Screenshot 2024-07-09 195035](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175048513/befca06d-4ab8-41fa-adf7-10c84d0ff819)

**Comment by JedK721** [comment]
**Wits-i**

Incessant attempts:
![Screenshot 2024-07-09 200735](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/98274272/934f7dfd-3cbf-4f45-b238-1d968655a6ec)

**Comment by MishkaSingh1** [comment]
**EEElite (UCT)**
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175049735/f6f03dcd-8333-4d01-b615-04d8b356a104)

**Comment by hoperiley0806** [comment]
**Parallel Pioneers**
"sudo journalctl | grep sshd" screenshot

![WhatsApp Image 2024-07-09 at 22 43 09_4d8499c6](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/135278342/b37b3c34-4750-456c-9b40-4bd9431c9e28)

**Comment by Dc-Spears** [comment]
**CPUT! (DC Spears)**

_journalctl of sshd before "bruteforce"_
![cput-firewall-before](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175037009/871893a1-1dc4-460c-9a47-dab82f34b436)

_journalctl of sshd after "bruteforce"_
![cput-firewall-after](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175037009/078555de-7b17-46dd-8572-618ad57c48d3)

**Comment by Rosemary2301** [comment]
Algorithm Avengers

![Screenshot (7)](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/143855810/e786083b-05b3-4a03-89a2-2ffc468837e6)

**Comment by siyabonga-mashilwane** [comment]
### **ThunderBytes**
![Screenshot 2024-07-09 172248](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/81867821/434096f9-e8a7-4029-99cf-a0e47d9d1767)

**Comment by SfisoMncube** [comment]
**Khulisa IP-A** 

![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/151764841/46c9b85a-d348-45c3-a3e4-545668b03669)

**Comment by iamyou-uoymai** [comment]
![journalctl of sshd](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/120018440/aa872140-9bed-47c3-9e82-f469e8812721)

**Comment by 15kay** [comment]
HYPERNET

sudo journalctl | grep sshd

![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/94864138/5d6a4ac8-3afb-4e8c-988f-684cba500b4e)

**Comment by mbhonichau** [comment]
WSU ELITE

![Screenshot 2024-07-10 162407](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/160317954/782b08ec-c5fa-40f1-8b76-8d36803070d6)

**Comment by ADELAIDEMOGALE** [comment]
### **Khulisa Data**

### **sudo journalctl | grep sshd**

![Screenshot (692)](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/166052302/00fd7a84-3c98-4e3c-9609-f79cdef5147f)

**Comment by ndumezulu-ngwenya** [comment]
![Screenshot 2024-07-10 171131](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/89075456/abab8185-a102-4d1d-9372-eca24736faab)
UNISA-SCT'24

sudo journalctl | grep sshd

**Comment by lisabeeaan** [comment]
**HPC Wizards**
![image](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/136716760/6f1953f4-8218-4ac2-bb25-b1d1e0117087)

**Comment by SizweMarole-21389** [comment]
Cybershield 

![Stateful_firewall](https://github.com/chpc-tech-eval/chpc24-scc-nmu/assets/175023186/0c185b93-5379-4575-8aa4-c34f70e51c5d)

**Comment by KayGee-17** [comment]
sudo journalctl | grep sshd

<img width="959" alt="incessant" src="https://github.com/user-attachments/assets/4db2fd77-ae54-456e-881f-66a66353f0f3">

**Comment by 15kay** [comment]
HYPERNET
![image](https://github.com/user-attachments/assets/516cf2e6-e7f0-42ce-bfa3-8beed63d55d6)

**Comment by KamvaMj** [comment]
DC Daggers

![Screenshot (73)](https://github.com/user-attachments/assets/80629fff-3851-42f7-a64a-663f959b6f9c)

**Comment by ShavirV** [comment]
TuksByters 
![image](https://github.com/user-attachments/assets/675533b4-504c-4079-a259-07538ccf728b)

**Comment by Cloud-Strive** [comment]
runtime blinders

![Screenshot (3)](https://github.com/user-attachments/assets/ac1345ec-50f5-47e3-aae7-fd7979babe45)

**Comment by Abahle947** [comment]
Techtribe  sudo journalctl | grep sshd
![thumbnail_PHOTO-2025-07-08-21-55-45](https://github.com/user-attachments/assets/c6b54f32-3df5-42ba-9add-09127500ada4)

**Comment by mdvar2** [comment]
![Screenshot 2025-07-08 205432](https://github.com/user-attachments/assets/f5b8fc9b-f74c-4ab3-989e-80dc49582c0c)
#Byte_Force

**Comment by Cloud-Strive** [comment]
runtime blinders

htop
![Screenshot (4)](https://github.com/user-attachments/assets/9cff9003-43e7-49a8-b186-65daa0e03643)

btop
![Screenshot (5)](https://github.com/user-attachments/assets/6d98ff97-8949-49ef-9b80-200e25d985a0)

**Comment by Moagi99** [comment]
TUT_X

![Screenshot (12)](https://github.com/user-attachments/assets/6cd68cb0-fb96-4ae8-bd81-8d223f7d9968)

**Comment by MphephuSamuel** [comment]
computer champions
![image](https://github.com/user-attachments/assets/4952a166-b29e-4b99-aafc-37cc53c3733a)

**Comment by Swissisking21** [comment]
Techno geeks
![Screenshot (283)](https://github.com/user-attachments/assets/b07333cb-53e0-4774-9b98-83390d4232b9)

**Comment by Shank-RTX** [comment]
OnePC
![Screenshot (279)](https://github.com/user-attachments/assets/802d5832-9660-46e3-8a13-018c618cf3c3)

**Comment by UNISA-SCT25** [comment]
![journastcl](https://github.com/user-attachments/assets/8f59a5c8-3a93-4396-8644-60f6306ecb6e)

**Comment by MubeenD1** [comment]
wits-1
said incessant attempts:
![image](https://github.com/user-attachments/assets/609719e6-82c3-4ca1-a78b-d2f1f4b4365d)

**Comment by Kevin-Ebrahim** [comment]
![Screenshot 2025-07-09 212252](https://github.com/user-attachments/assets/6cc1aea7-cdce-4819-b26e-d66aa2500ff1)
Wits-A

**Comment by SFBdragon** [comment]
0v1rcl0ck1d

![Screenshot_20250709_230917](https://github.com/user-attachments/assets/4a0e2829-e540-40f1-8a23-7e1712643258)

**Comment by Cloud-Strive** [comment]
Runtime Blinders
sudo journalctl | grep sshd


<img width="1920" height="1080" alt="Screenshot (15)" src="https://github.com/user-attachments/assets/e48a0209-5cd6-4ead-a55a-cdffd1b0eb1f" />

**Comment by omo-bit** [comment]
Wits-1
sudo journalctl | grep sshd

<img width="1453" height="662" alt="image" src="https://github.com/user-attachments/assets/2fc1855f-9023-4b94-af6f-fcea7e2bd0c9" />

**Comment by RoarkeBailey** [comment]
<img width="1917" height="1076" alt="Screenshot 2026-07-06 202059" src="https://github.com/user-attachments/assets/89230910-057a-4f07-9078-c8a9e6144658" />

G3MIKNIGHTS 
sudo journalctl | grep sshd

**Comment by ShavirV** [comment]
<img width="1918" height="1079" alt="image" src="https://github.com/user-attachments/assets/374e48bb-bf74-4058-9ff8-37a8101cba5b" />

UPtime
journalctl | grep sshd

**Comment by CadmusKei** [comment]
Team Jargon (UWC) | SCC 2026 | journalctl for Bruteforce SSH Attempts
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/3b7e5fc7-95c4-4824-a87c-4cc0a75d617a" />

**Comment by saxs-14** [comment]
3ncrypt3d (UMP)
sudo journalctl | grep sshd

<img width="1918" height="1078" alt="image" src="https://github.com/user-attachments/assets/c0f66bae-fff6-4d28-b68f-8e7c459743b0" />

**Comment by ThapeloVundla** [comment]
DC Spears | running sudo journalctl | grep sshd

<img width="1860" height="999" alt="Screenshot from 2026-07-07 10-21-11" src="https://github.com/user-attachments/assets/568cf9a2-8b7c-40fe-bba2-9559c1d21297" />

**Comment by bytedebuggers-dotcom** [comment]
Byte Debuggers (UFH)
**Running sudo journalctl | grep sshd**
<img width="1905" height="1046" alt="Screenshot from 2026-07-07 15-36-59" src="https://github.com/user-attachments/assets/97b0760c-3a4c-4fc1-9836-f08a10a29117" />

**Comment by Slindokuhle19** [comment]
UnizuluComputeX
sudo journalctl | grep ssh
<img width="1816" height="939" alt="Screenshot from 2026-07-07 17-52-49" src="https://github.com/user-attachments/assets/38e8720a-55df-4dd9-892e-724d2e32c8a4" />

**Comment by s225875411** [comment]
MadibazBits

<img width="1850" height="998" alt="image (2)" src="https://github.com/user-attachments/assets/9cd15da1-1096-4bf2-8449-f6de8fe69928" />

**Comment by K659** [comment]
Resilient Geeks (UL) ~  sudo journalctl | grep ssh
<img width="1871" height="1015" alt="image" src="https://github.com/user-attachments/assets/504b8eec-ccb9-412f-a733-2887a333d93c" />

**Comment by LufunoMiles** [comment]
Digital Dynamos (University of Venda) sudo journalctl | grep <img width="1844" height="952" alt="image" src="https://github.com/user-attachments/assets/04c04659-ffb4-47a5-903b-8b92b9c7a83f" />

**Comment by mabotenick97-cloud** [comment]
CyberOps

<img width="917" height="967" alt="Screenshot from 2026-07-07 15-19-22" src="https://github.com/user-attachments/assets/0fa8189f-6c40-4c6d-bca0-2a8290a789ac" />

**Comment by JayJay-lab** [comment]
Sol_Plaatje_A   SCC 2026       
<img width="1895" height="1069" alt="image" src="https://github.com/user-attachments/assets/ad846edb-7bd9-4c3a-90fb-0522a15215bd" />

**Comment by mpho21925** [comment]
Ampere Alliance | SCC 2026
sudo journalctl | grep sshd 

<img width="1920" height="1080" alt="Screenshot 2026-07-08 090713" src="https://github.com/user-attachments/assets/215e0f5f-6217-445e-b292-1ac2831b901e" />

**Comment by Leo-OB** [comment]
## DC Axes (CPUT) | SCC 2026
### Outputs of `sudo journalctl | grep sshd`
<img width="1915" height="1077" alt="T2_SudoJournalGrep" src="https://github.com/user-attachments/assets/297600f9-84b7-464e-b445-d0798af03204" />

**Comment by Lihle0512** [comment]
Chaos Compilers

<img width="1858" height="960" alt="Screenshot from 2026-07-09 08-29-31" src="https://github.com/user-attachments/assets/9814a37e-4a3a-4ef1-88bc-33372de95722" />

**Comment by switchsquad** [comment]
<img width="1860" height="961" alt="image" src="https://github.com/user-attachments/assets/9ab75f04-b021-468f-a137-64568e011d61" />
Switch Squad(TUT)

**Comment by majaha8** [comment]
<img width="935" height="1011" alt="Stateful Firewall and Bruteforce SSH Attempts
" src="https://github.com/user-attachments/assets/7a275a7d-46ba-4acc-9046-c68653a83bb4" />

**Comment by majaha8** [comment]
Team Nebula|SCC 2026

**Comment by BleSSednav3** [comment]
Resilient Geeks 
<img width="1865" height="1042" alt="image" src="https://github.com/user-attachments/assets/486d2589-3c96-4705-a05e-2aac0d8c23be" />

**Comment by Leeu-Fusi** [comment]
Cluster Commanders | CUT

<img width="1801" height="1005" alt="Screenshot from 2026-07-10 14-26-38" src="https://github.com/user-attachments/assets/f0bfc662-dd72-40c6-bafb-672f17f0c4bc" />

**Comment by MrLampies** [comment]
DC Spears ssh

<img width="1860" height="999" alt="Screenshot from 2026-07-07 10-21-11" src="https://github.com/user-attachments/assets/837d19d6-6a8f-4285-be91-993170c98f94" />

**Comment by TKSebogo62** [comment]
Ampere Alliance | SCC 2026
Stateful firewall
<img width="937" height="950" alt="image" src="https://github.com/user-attachments/assets/57cc2b1f-c562-4cc1-ad76-683b6bf73c22" />

**Comment by Slindokuhle19** [comment]
UnizuluComputeX
<img width="1816" height="939" alt="Screenshot from 2026-07-07 17-52-49" src="https://github.com/user-attachments/assets/699f8284-ca34-4317-bde4-80b60485ac32" />

**Comment by VerilM** [comment]
Wits-1:

<img width="1862" height="896" alt="image" src="https://github.com/user-attachments/assets/d4126744-192b-4e89-a9fb-13b34bcb8b2e" />
